src/components/Select/Select.tsx

```diff
@@ -44,45 +55,100 @@ const Select = React.forwardRef<HTMLInputElement, Props & InputProps>(
       selectedOption = emptyValue,
       multi = false,
       options,
+      isAsync = false,
+      asyncSearch = () => {},
       status = 'normal',
+      minCharactersToSearch = 0,
+      highlightSearch = false,
       ...restInputProps
     },
     ref
   ) => {
     const theme = useTheme();
     const [open, setOpen] = React.useState(false);
+    const [isLoading, setIsLoading] = React.useState(false);
     const [inputValue, setInputValue] = React.useState(defaultValue || selectedOption);
     const [searchValue, setSearchValue] = React.useState('');
 
     useEffect(() => {
       setInputValue(defaultValue || selectedOption);
     }, [defaultValue, selectedOption]);
 
+    useEffect(() => {
+      if (isAsync) {
+        setIsLoading(false);
+      }
+    }, [isAsync, options]);
+
     const handleOptionClick = (option: SelectOption) => {
       setInputValue(option);
       setOpen(false);
       setSearchValue('');
       handleSelectedOption(option);
     };
 
+    const handleOnKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
+      // if backspace
+      if (e.keyCode === 8) {
+        setInputValue(emptyValue);
+      }
+    };
+
+    const debouncedOnChange = React.useCallback(
+      debounce(term => {
```

Comment:
Maybe we should only debounce side effects on leading action to avoid unexpected behavior or delays, e.g. { leading: true }, what do you think?

Reply:
Nice idea! I'm adding it now

Reply:
These are the results with leading set to true and false. While in theory it sounded better, I believe that the default behavior results in a smoother experience for the user, even though they will have to wait for the debounce timer to end. I can actually reduce the timer from 500ms to 400ms.

leading: true
![leading](https://user-images.githubusercontent.com/13350837/109653257-79317900-7b69-11eb-8572-a9a76cce8f17.gif)

leading: false
![trailing](https://user-images.githubusercontent.com/13350837/109653260-79ca0f80-7b69-11eb-8338-2b9aa2c68583.gif)

Reply:
Thanks for the descriptive comment @whosyourtaco. Let's leave the implementation as is and take it offline for further investigation.